medusa/helpers.py

```diff
@@ -1189,7 +1189,19 @@ def get_url(url, post_data=None, params=None, headers=None, timeout=30, session=
     hooks, cookies, verify, proxies = request_defaults(kwargs)
     method = u'POST' if post_data else u'GET'
 
+    # Get RequestPolice instance object.
+    rpolice = kwargs.pop(u'rpolice', None)
+
+    from helper.request_police import RequestPoliceException
+
     try:
+        if rpolice:
+            try:
+                [rpolice_check() for rpolice_check in rpolice.enabled_police_request_hooks]
+            except RequestPoliceException as e:
```

Comment:
These types of checks for API limit, etc should be part of the service's session.  You should only use a response hook as a last resort.  A response hook can only be checked after the request has already been sent and only applies the condition if the hook has already been attached.  If you want to check before sending a request, a better approach would be to subclass requests.Request and implement the functionality there.